pkg/policy/background.go

```diff
@@ -25,6 +25,37 @@ func ContainsVariablesOtherThanObject(policy kyverno.ClusterPolicy) error {
 			return fmt.Errorf("invalid variable used at path: spec/rules[%d]/exclude/%s", idx, path)
 		}
 
+		if len(rule.MatchResources.Any) > 0 {
+			for _, value := range rule.MatchResources.Any {
+				if path := userInfoDefined(value.UserInfo); path != "" {
+					return fmt.Errorf("invalid variable used at path: spec/rules[%d]/match/%s", idx, path)
```

Comment:
Please update the error message with the precise path, i.e., `spec/rules[%d]/match/any[%d]`.